internal/cli/usercmd/push.go

```diff
@@ -268,24 +268,38 @@ func (c *EpinioClient) AppPush(ctx context.Context, manifest models.ApplicationM
 func (c *EpinioClient) uploadSources(log logr.Logger, appRef models.AppRef, source string) (string, error) {
 	c.ui.Normal().Msg("Collecting the application sources ...")
 
-	tmpDir, tarball, err := helpers.Tar(source)
-	defer func() {
-		if tmpDir != "" {
-			_ = os.RemoveAll(tmpDir)
-		}
-	}()
+	fileInfo, err := os.Stat(source)
 	if err != nil {
 		return "", err
 	}
 
+	var archive string
+
+	if fileInfo.IsDir() {
+		// package directory as archive/tarball
+		tmpDir, tarball, err := helpers.Tar(source)
+		defer func() {
+			if tmpDir != "" {
+				_ = os.RemoveAll(tmpDir)
+			}
+		}()
```

Comment:
if `tmpDir` is empty `""` the `os.RemoveAll` is not going to fail, so it's safe to call it anyway. Also we are not even logging the error, so we can just call it:

```suggestion
		defer os.RemoveAll(tmpDir)
```

Reply:
Done. Commit 450458216cfa6b5b756f082194f3a26a246a33c9